### PR TITLE
Only query related datasets locked status in SharingTaskService

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/zfs.py [hunk 1]
```diff
@@ -474,24 +494,40 @@ def query(self, filters=None, options=None):
 
         with libzfs.ZFS() as zfs:
             # Handle `id` filter specially to avoiding getting all datasets
-            if filters and len(filters) == 1 and list(filters[0][:2]) == ['id', '=']:
+            if filters and filters[0][0] == 'id' and filters[0][1] in ['=', 'in']:
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/zfs.py:497]: `['id', 'in']`

Comment (william-gr) [reply src/middlewared/middlewared/plugins/zfs.py:497]: No, nvm

**Review by william-gr — COMMENTED** [review]
This seems way more intrusive than what I was hoping for 12.0, unfortunately

**Comment by themylogin** [comment]
@william-gr unfortunately, this is even more intrusive that it seems, libzfs behavior is different for `zfs.get_dataset(...).__getstate__())` and `zfs.datasets_serialized`, we are now fixing at least one difference we've found. I was thinking maybe we backport this to SCALE (performance improvement is really very good) and in 12.0 fix one specific iscsi extent endpoint @rkojedzinszky is having performance issues with?

**Comment by william-gr** [comment]
@themylogin I agree

**Comment by rkojedzinszky** [comment]
@themylogin @william-gr thank you for your great efforts.

**Pull request closed without merging**